src/cross_section.F90

```diff
@@ -140,169 +133,160 @@ end subroutine calculate_xs
+        if (nuc % fissionable) then
+          micro_xs(i_nuclide) % fission = sigF
+          micro_xs(i_nuclide) % nu_fission = sigF * nuc % nu(E, EMISSION_TOTAL)
         else
-          ! Determine bounding indices based on which equal log-spaced interval
-          ! the energy is in
-          i_low  = nuc % grid_index(i_log_union)
-          i_high = nuc % grid_index(i_log_union + 1) + 1
-
-          ! Perform binary search over reduced range
-          i_grid = binary_search(nuc % energy(i_low:i_high), &
-               i_high - i_low + 1, E) + i_low - 1
+          micro_xs(i_nuclide) % fission    = ZERO
+          micro_xs(i_nuclide) % nu_fission = ZERO
         end if
 
-      case (GRID_NUCLIDE)
-        ! Perform binary search on the nuclide energy grid in order to determine
-        ! which points to interpolate between
-
-        if (E <= nuc % energy(1)) then
-          i_grid = 1
-        elseif (E > nuc % energy(nuc % n_grid)) then
-          i_grid = nuc % n_grid - 1
-        else
-          i_grid = binary_search(nuc % energy, nuc % n_grid, E)
-        end if
-
-      end select
+        ! Ensure these values are set
+        ! Note, the only time either is used is in one of 4 places:
+        ! 1. physics.F90 - scatter - For inelastic scatter.
+        ! 2. physics.F90 - sample_fission - For partial fissions.
+        ! 3. tally.F90 - score_general - For tallying on MTxxx reactions.
+        ! 4. cross_section.F90 - calculate_urr_xs - For unresolved purposes.
+        ! It is worth noting that none of these occur in the resolved
+        ! resonance range, so the value here does not matter.
+        micro_xs(i_nuclide) % index_temp    = i_temp
+        micro_xs(i_nuclide) % index_grid    = 0
+        micro_xs(i_nuclide) % interp_factor = ZERO
+      else
+        associate (grid => nuc % grid(i_temp), xs => nuc % sum_xs(i_temp))
+          ! Determine the energy grid index using a logarithmic mapping to reduce
+          ! the energy range over which a binary search needs to be performed
+
+          if (E < grid % energy(1)) then
+            i_grid = 1
+          elseif (E > grid % energy(size(grid % energy))) then
+            i_grid = size(grid % energy) - 1
+          else
+            ! Determine bounding indices based on which equal log-spaced interval
+            ! the energy is in
+            i_low  = grid % grid_index(i_log_union)
+            i_high = grid % grid_index(i_log_union + 1) + 1
+
+            ! Perform binary search over reduced range
+            i_grid = binary_search(grid % energy(i_low:i_high), &
+                 i_high - i_low + 1, E) + i_low - 1
+          end if
 
-      ! check for rare case where two energy points are the same
-      if (nuc % energy(i_grid) == nuc % energy(i_grid+1)) i_grid = i_grid + 1
+          ! check for rare case where two energy points are the same
+          if (grid % energy(i_grid) == grid % energy(i_grid + 1)) &
+               i_grid = i_grid + 1
```

Comment:
Now that we control the data generation should we just chcek for this on the python side?
(Can you think of anything else like this we can fix now?)